components/sidebar/add_channels_cta_button.tsx

```diff
@@ -0,0 +1,159 @@
+    const canJoinPublicChannel = useSelector((state: GlobalState) => haveICurrentChannelPermission(state, Permissions.JOIN_PUBLIC_CHANNELS));
+    const isAddChannelCtaOpen = useSelector(isAddChannelCtaDropdownOpen);
+    const currentUserId = useSelector(getCurrentUserId);
+    const openAddChannelsCtaOpen = useCallback((open: boolean) => {
+        dispatch(setAddChannelCtaDropdown(open));
+    }, []);
+
+    let buttonClass = 'SidebarChannelNavigator__addChannelsCtaLhsButton';
+
+    if (!touchedAddChannelsCtaButton) {
+        buttonClass += ' SidebarChannelNavigator__addChannelsCtaLhsButton--untouched';
+    }
+
+    if (!currentTeamId) {
+        return null;
+    }
+
+    const showMoreChannelsModal = () => {
+        dispatch(openModal({
+            modalId: ModalIdentifiers.MORE_CHANNELS,
+            dialogType: MoreChannels,
+            dialogProps: {morePublicChannelsModalType: 'public'},
+        }));
+        trackEvent('ui', 'browse_channels_button_is_clicked');
+    };
+
+    const showNewChannelModal = () => {
+        dispatch(openModal({
+            modalId: ModalIdentifiers.NEW_CHANNEL_MODAL,
+            dialogType: NewChannelModal,
+        }));
+        trackEvent('ui', 'create_new_channel_button_is_clicked');
+    };
+
+    const renderDropdownItems = () => {
+        let joinPublicChannel;
+        if (canJoinPublicChannel) {
+            joinPublicChannel = (
+                <Menu.ItemAction
+                    id='showMoreChannels'
+                    onClick={showMoreChannelsModal}
+                    text={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.browseChannels', defaultMessage: 'Browse Channels'})}
+                />
+            );
+        }
+
+        let createChannel;
+        if (canCreateChannel) {
+            createChannel = (
+                <Menu.ItemAction
+                    id='showNewChannel'
+                    onClick={showNewChannelModal}
+                    text={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.createNewChannel', defaultMessage: 'Create New Channel'})}
+                />
+            );
+        }
+
+        return (
+            <>
+                <Menu.Group>
+                    {createChannel}
+                    {joinPublicChannel}
+                </Menu.Group>
+            </>
+        );
+    };
+
+    const trackOpen = (opened: boolean) => {
+        openAddChannelsCtaOpen(opened);
+        trackEvent('ui', 'add_channels_cta_button_clicked');
+        if (!touchedAddChannelsCtaButton) {
+            dispatch(savePreferences(
+                currentUserId,
+                [{
+                    category: Preferences.TOUCHED,
+                    user_id: currentUserId,
+                    name: Touched.ADD_CHANNELS_CTA,
+                    value: 'true',
+                }],
+            ));
+        }
+    };
+
+    if (!canCreateChannel && !canJoinPublicChannel) {
```

Comment:
Nit: Might make sense to group this early return next to the other one up above.

Reply:
+1